### PR TITLE
ikev1: add ikev1 test - v8

### PR DESCRIPTION
**Pull request by satta**
Previous PR: https://github.com/OISF/suricata-verify/pull/351

* Only rebased so far.

### CODE REVIEW

File: tests/ikev1-rules/test.rules [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+alert ike any any -> any any (msg:"ike server key exchange"; ike.key_exchange_payload; content:"|6d026d5616c45be05e5b898411e9f95d195cea009ad22c62bef06c571b7cfbc4792f45564ec710ac584aa18d20cbc8f5f8910666b89e4ee2f95abc0230e2cba1b88ac4bba7fcc818a986c01a4ca865a5eb82884dbec85bfd7d1a303b09894dcf2e3785fd79dba225377cf8cca009ceffbb6aa38b648c4b05404f1cfaac361aff|"; flow:to_client; sid:13;)
+alert ike any any -> any any (msg:"ike client key exchange"; ike.key_exchange_payload; content:"|3504d3d2ed14e0ca03b851a51a9da2e5a4c14c1d7ec3e1fbe950025424514b3c69ed7fbb44e09225da52d2a92604a99bf61b7beed7fbfa635e82f065f4fe780751354dbe474c3de7207dcf69fdbbed32c1691cc149b318eee00370e65fc3069bbacfb013467173966e9d5f4bc4f3857e359bba3adbb6efeea516f3897d8534f3|"; flow:to_server; sid:14;)
+alert ike any any -> any any (msg:"ike key payload length"; ike.key_exchange_payload_length:>100; sid:9;)
+alert ike any any -> any any (msg:"ike nonce payload length"; ike.nonce_payload_length:<200; sid:3;)
+alert ike any any -> any any (msg:"ike nonce payload"; ike.nonce_payload; content:"|89d7c8fbf94b515b521d5d9589c2602021e1a709|"; sid:4;)
```
Comment (victorjulien) [inline tests/ikev1-rules/test.rules:16]: for each of the new sticky buffers I'd like to see some rules with transforms as well to validate that this works _and_ that we don't have interference between non-transformed and transformed buffers as we've had with file_data.

**Comment by victorjulien** [comment]
The failure on the 6 branch will have to be addressed. I don't want to reduce the testing done on that branch, so we'll need checks specifying the correct versions.

**Comment by satta** [comment]
I'll see to it and make sure we have tests for both versions.

File: tests/ikev1-rules/test.rules (hunk 1 above)
Comment (satta) [reply tests/ikev1-rules/test.rules:16]: Sure, I will add rules that trigger on transformed buffer contents for some sensible transforms.

**Comment by satta** [comment]
Next PR: #472

**Pull request closed without merging**